pattern_library/management/commands/render_patterns.py

```diff
@@ -0,0 +1,77 @@
+from pathlib import Path
+
+from django.core.management.base import BaseCommand
+from django.test.client import RequestFactory
+
+from pattern_library.utils import (
+    get_pattern_templates, render_pattern
+)
+
+
+class Command(BaseCommand):
+    help = "Renders all django-pattern-library patterns to HTML files, in a directory structure."
+
+    def add_arguments(self, parser):
+        super().add_arguments(parser)
+        parser.add_argument(
+            '--output',
+            '-o',
+            action='store',
+            dest='output_dir',
+            default='dpl-rendered-patterns',
+            help='Directory where to render your patterns',
+            type=str,
+        )
+        parser.add_argument(
+            '--dry-run',
+            action='store_true',
+            help="Render the patterns without writing them to disk.",
+        )
+
+    def handle(self, **options):
+        self.verbosity = options['verbosity']
+        self.dry_run = options['dry_run']
+        self.output_dir = options['output_dir']
+
+        templates = get_pattern_templates()
+
+        factory = RequestFactory()
+        request = factory.get('/')
+
+        if self.verbosity >= 2:
+            self.stdout.write(f'Target directory: {self.output_dir}')
```

Comment:
I did it like this because I wanted stderr to be left for warnings and errors from rendering the templates. I need to check where those go though (warnings are stderr, errors I haven’t checked)

Reply:
The handling of errors/warnings from `logging` calls is going to vary project to project depending on the `LOGGING` setting among other things. I'd be more concerned about keeping logging output of `stdout` than mixing it with other logging, but I don't want to hold this up unnecessarily by nit-picking.